### PR TITLE
Add flexibility to enable required repos

### LINKED ISSUE #305: Have a better error msg when Satellite repo is not enabled
When running sat-clone w/ manual registration of repos, we can do a better job checking for Satellite repo and instructing users to manually setup the repos if they are not enabled. Currently yum errors out if repos are not enabled.

### PR DESCRIPTION
**Pull request by sthirugn**
Fixes #305

### CODE REVIEW

File: satellite-clone-vars.sample.yml [hunk 1]
```diff
@@ -28,9 +28,8 @@
 # uncomment the following line to override this option (defaults to true)
 #run_pre_install_check: true
 
-# Use register_to_portal to automatically register to Red Hat Portal. By default,
-#   - the host will not be registered to RH portal.
-#   - it is the responsibility of the user to make the required repos available.
+# Use register_to_portal to automatically register to Red Hat Portal. By default,the host will not
```
Comment (johnpmitsch) [inline satellite-clone-vars.sample.yml]: looks like a space is missing `default,the host`

Comment (johnpmitsch) [reply satellite-clone-vars.sample.yml]: host is also confusing here (makes me think of a Satellite host), could we change that to "system"?

File: roles/satellite-clone/tasks/main.yml [hunk 2]
```diff
@@ -49,12 +49,12 @@
 - name: register host
   command: "subscription-manager register --force --activationkey='{{ activationkey }}' --org '{{ org }}'"
   when: register_to_portal
+
+# Enable Required repos
```
Comment (johnpmitsch) [inline roles/satellite-clone/tasks/main.yml]: nitpick, but this is redundant since the task name is saying the same thing.

**Comment by sthirugn** [comment]
@johnpmitsch updated

**Comment by johnpmitsch** [comment]
If this is the new default I think there are two more things we have to do with this change:
1) Update the documentation to include attaching the correct satellite subscription to the machine
2) Add a check that the subscription is attached or at the least catch the errors when the correct sub isn't attached and display a proper error message.

Currently, I spin up a system, set up sat-clone, backup, config file, etc... but don't attach subs, this is the error I get:

```
TASK [satellite-clone : Enable required repos for Satellite installation] ********************************************************************
Friday 26 January 2018  02:04:08 +0000 (0:00:09.574)       0:00:11.454 ******** 
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["subscription-manager", "repos", "--enable", "rhel-7-server-rpms", "--enable", "rhel-server-rhscl-7-rpms", "--enable", "rhel-7-server-satellite-6.2-rpms"], "delta": "0:00:08.162395", "end": "2018-01-26 02:04:16.728418", "msg": "non-zero return code", "rc": 1, "start": "2018-01-26 02:04:08.566023", "stderr": "", "stderr_lines": [], "stdout": "Error: 'rhel-7-server-satellite-6.2-rpms' does not match a valid repository ID. Use \"subscription-manager repos --list\" to see valid repositories.\nRepository 'rhel-7-server-rpms' is enabled for this system.\nRepository 'rhel-server-rhscl-7-rpms' is enabled for this system.", "stdout_lines": ["Error: 'rhel-7-server-satellite-6.2-rpms' does not match a valid repository ID. Use \"subscription-manager repos --list\" to see valid repositories.", "Repository 'rhel-7-server-rpms' is enabled for this system.", "Repository 'rhel-server-rhscl-7-rpms' is enabled for this system."]}

cmd: 
subscription-manager
repos
--enable
rhel-7-server-rpms
--enable
rhel-server-rhscl-7-rpms
--enable
rhel-7-server-satellite-6.2-rpms

start: 2018-01-26 02:04:08.566023

end: 2018-01-26 02:04:16.728418

delta: 0:00:08.162395

msg: non-zero return code

stdout: Error: 'rhel-7-server-satellite-6.2-rpms' does not match a valid repository ID. Use "subscription-manager repos --list" to see valid repositories.
Repository 'rhel-7-server-rpms' is enabled for this system.
Repository 'rhel-server-rhscl-7-rpms' is enabled for this system.
	to retry, use: --limit @/root/satellite-clone/satellite-clone-playbook.retry

PLAY RECAP ***********************************************************************************************************************************
localhost                  : ok=23   changed=3    unreachable=0    failed=1   

```

It isn't immediately obvious what the issue is and we could catch this and display a proper error msg, or have a check in the pre-checks for a correctly attached subscription.

File: roles/satellite-clone/defaults/main.yml [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-activationkey: changeme
-org: changeme
 backup_dir: /backup
 disable_firewall: False
 run_katello_reindex: False
 run_pre_install_check: True
 register_to_portal: False
+activationkey: changeme
+org: changeme
+enable_repos: True
```
Comment (sthirugn) [inline roles/satellite-clone/defaults/main.yml]: Note that the order is changed to match the sample var file.

**Comment by sthirugn** [comment]
Added new logic for enable repos. This is how it works now.

if user forgot to register to Red Hat portal or do not have right subscriptions:
```sh
TASK [satellite-clone : Check that the repos are disabled without errors] ********************************************************************************************
Friday 26 January 2018  14:48:11 -0500 (0:00:03.868)       0:00:03.941 ******** 
fatal: [ibm-x3250m4-07.lab.eng.rdu2.redhat.com]: FAILED! => {"changed": false, "failed": true, "msg": "Disabling repos failed. Make sure that your system is registered to Red Hat to get contents. Alternatively, you can skip this task by specifying `enable_repos` to `False`."}

msg: Disabling repos failed. Make sure that your system is registered to Red Hat to get contents. Alternatively, you can skip this task by specifying `enable_repos` to `False`.
```

if user did not have required repos for satellite installation:

```sh
TASK [satellite-clone : Check that the repos are enabled without errors] *********************************************************************************************
Friday 26 January 2018  15:02:50 -0500 (0:00:08.453)       0:00:08.512 ******** 
fatal: [ibm-x3250m4-07.lab.eng.rdu2.redhat.com]: FAILED! => {"changed": false, "failed": true, "msg": "Enabling required repos failed.  Make sure that your system has access to the following repos: rhel-7-server-rpms, rhel-server-rhscl-7-rpms, rhel-7-server-satellite-6.2-rpms"}

msg: Enabling required repos failed.  Make sure that your system has access to the following repos: rhel-7-server-rpms, rhel-server-rhscl-7-rpms, rhel-7-server-satellite-6.2-rpms
```

File: roles/satellite-clone/tasks/main.yml [hunk 4]
```diff
@@ -45,16 +45,33 @@
...
-- name: disable all repos
+
+- name: Disable all repos
   command: subscription-manager repos --disable "*"
-  when: register_to_portal
-- name: Enable required repos
+  ignore_errors: True
+  register: disable_repos_result
+  when: enable_repos
+- name: Check that the repos are disabled without errors
```
Comment (johnpmitsch) [inline roles/satellite-clone/tasks/main.yml]: please space out each separate tasks

File: docs/satellite-clone.md [hunk 5]
```diff
@@ -29,10 +29,11 @@
    ```console
      # cp satellite-clone-vars.sample.yml satellite-clone-vars.yml
    ```
-2. Place the backup files in a folder on the blank machine. This folder path is specified in the `backup_dir` variable in `satellite-clone-vars.yml`.
-3. Update the folder path of the backup files using the `backup_dir` variable in `satellite-clone-vars.yml`.
-4. If you are cloning RHEL 6 backup data to a RHEL 7 machine, update the variable `rhel_migration` to true in `satellite-clone-vars.yml`.
-5. Run the ansible playbook from the root directory of this project:
+2. Place the backup files in `/backup` folder on the blank machine. If using a different folder, update `backup_dir` variable in `satellite-clone-vars.yml`.
+3. If you are cloning RHEL 6 backup data to a RHEL 7 machine, update the variable `rhel_migration` to true in `satellite-clone-vars.yml`.
+4. You are required to register and subscribe the blank machine to Red Hat portal to get contents for Satellite installation.  Alternatively, you can override `register_to_portal` to `true` and update `activationkey`, `org` variables in `satellite-clone-vars.yml`.
```
Comment (johnpmitsch) [inline docs/satellite-clone.md]: I think the second sentence can be clearer on what that actually does. Why would I change all those variables?
A suggestion:
>Alternatively, you use an activation key from the Red Hat portal to automatically set up the correct subscriptions for you. To do this, override `register_to_portal` to `true`, then update `activationkey` and `org` variables in `satellite-clone-vars.yml`.

File: roles/satellite-clone/tasks/main.yml [hunk 6]
```diff
@@ -45,16 +45,33 @@
...
+  fail:
+    msg: "Disabling repos failed. Make sure that your system is registered and subscribed to Red Hat. Alternatively,
+         you can skip this task by specifying `enable_repos` to `False`."
```
Comment (johnpmitsch) [inline roles/satellite-clone/tasks/main.yml]: A little more info in this error message (and the one below it) couldn't hurt. 
>Alternatively, if you would like to enable the Satellite repos yourself, you can skip this task by specifying `enable_repos` to `False` "

File: roles/satellite-clone/tasks/main.yml [hunk 7]
```diff
@@ -45,16 +45,33 @@
...
+  when: enable_repos and disable_repos_result.rc != 0
```
Comment (johnpmitsch) [inline roles/satellite-clone/tasks/main.yml]: I think you can do `disable_repos_result | success` in ansible now, which is a bit more readable (same for below too)

Comment (johnpmitsch) [reply roles/satellite-clone/tasks/main.yml]: (it may need parenthesis to work with this boolean logic, not sure)

**Comment by sthirugn** [comment]
Review comments updated, I will have to look at the return code comment later.

**Comment by johnpmitsch** [comment]
yee-haw

**Comment by johnpmitsch** [comment]
yee-haw

File: docs/satellite-clone.md [hunk 8]
```diff
@@ -29,10 +29,11 @@
    ```console
      # cp satellite-clone-vars.sample.yml satellite-clone-vars.yml
    ```
-2. Place the backup files in a folder on the blank machine. This folder path is specified in the `backup_dir` variable in `satellite-clone-vars.yml`.
-3. Update the folder path of the backup files using the `backup_dir` variable in `satellite-clone-vars.yml`.
-4. If you are cloning RHEL 6 backup data to a RHEL 7 machine, update the variable `rhel_migration` to true in `satellite-clone-vars.yml`.
-5. Run the ansible playbook from the root directory of this project:
+2. Place the backup files in `/backup` folder on the blank machine. If using a different folder, update `backup_dir` variable in `satellite-clone-vars.yml`.
+3. If you are cloning RHEL 6 backup data to a RHEL 7 machine, update the variable `rhel_migration` to true in `satellite-clone-vars.yml`.
+4. You are required to register and subscribe the blank machine to Red Hat portal to get contents for Satellite installation.  Alternatively, to let the clone tool register to Red Hat portal, you can override `register_to_portal` to `true` and update `activationkey`, `org` variables in `satellite-clone-vars.yml`.
```
Comment (johnpmitsch) [inline docs/satellite-clone.md]: should be `get content for the Satellite installation`

**Comment by sthirugn** [comment]
Updated facts and the minor doc comment.

**Comment by johnpmitsch** [comment]
yee-haw

**Comment by johnpmitsch** [comment]
I think we should specify that Satellite subscriptions should be attached to the machine and some minor tweaks in the wording:

> Configuring repos failed. Make sure that your system is registered and subscribed to Red Hat and has the appropriate Satellite subscriptions attached. Alternatively, you can skip this task by specifying `enable_repos` to `False` if you would like to enable the appropriate Satellite repositories manually.

**Comment by johnpmitsch** [comment]
yee-haw

**Comment by johnpmitsch** [comment]
will merge when jenkins job is finished

**Comment by sthirugn** [comment]
rebased with master and squashed commits

**Comment by johnpmitsch** [comment]
thanks @sthirugn, awesome change!

**Pull request merged**